### PR TITLE
strip bootstrap repos and remove EOL OSes

### PR DESCRIPTION
**Pull request by mcalmer**
## What does this PR change?

- strip bootstrap repos to contain only venv-salt-minion and dependencies
- remove OS definitions which do not have the salt bundle

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- TBD

- [ ] **DONE**

## Test coverage
- No tests: already covered

- [x] **DONE**

## Links

Issue: https://github.com/SUSE/spacewalk/issues/19896

- [x] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

# Before you merge

Check [How to branch and merge properly](https://github.com/uyuni-project/uyuni/wiki/How-to-branch-and-merge-properly)!

### CODE REVIEW

**Comment by srbarrios** [comment]
It looks something to be well tested before merging.
If you have the opportunity to run this in a uyuni-pr test pipeline, that would be great.

**Comment by admd** [comment]
@deneb-alpha @m-czernek can you please take a look here tomorrow.

File: susemanager/susemanager.changes.mc.shrink-bootstrap-repos [hunk 1]
```diff
@@ -0,0 +1,2 @@
+- strip bootstrap repos to contain only venv-salt-minion and dependencies
+- remove OS definitiions which do not have the salt bundle
```
Comment (m-czernek) [inline susemanager/susemanager.changes.mc.shrink-bootstrap-repos:2]:
```suggestion
- remove OS definitions which do not have the salt bundle
```

**Review by m-czernek — APPROVED** [review]
From my POV, this is OK, though I'm unable to verify all package changes, e.g. not sure why we're keeping `gnupg` on Debian but not on Ubuntu. I can see that we require it in https://build.opensuse.org/projects/systemsmanagement:saltstack:bundle:next/packages/venv-salt-minion/files/venv-salt-minion.spec?expand=1; I assume that's because of old versions on the debian systems?

**Comment by mcalmer** [comment]
> From my POV, this is OK, though I'm unable to verify all package changes, e.g. not sure why we're keeping `gnupg` on Debian but not on Ubuntu. I can see that we require it in https://build.opensuse.org/projects/systemsmanagement:saltstack:bundle:next/packages/venv-salt-minion/files/venv-salt-minion.spec?expand=1; I assume that's because of old versions on the debian systems?

Debian does not have it installed by default. Don't ask me why.

**Comment by mcalmer** [comment]
> It looks something to be well tested before merging. If you have the opportunity to run this in a uyuni-pr test pipeline, that would be great, it will not cover all the products like in BV... but at least we will cover one.

That pipeline is currently not due to containers

**Review by m-czernek — APPROVED** [review]
Ok, understood. +1 from my side.

File: susemanager/src/mgr_bootstrap_data.py [hunk 2]
```diff
@@ -14,93 +14,7 @@
...
-    "libyaml",
-    "python-chardet",
-    "python-six",
-    "yum-plugin-security",
-    "yum",
-    "rpm-python",
-    "redhat-rpm-config",
-    "openssh-clients",
-]
+PKGLIST12 = ["logrotate", "libsqlite3-0", "venv-salt-minion"]
```
Comment (deneb-alpha) [inline susemanager/src/mgr_bootstrap_data.py:17]: Can we use a similar  syntax harmonizing  how we write  the list?
If we use something like this, if/when we will have to ad/remove packages, the diff should be less messy from  what I remember.

```suggestion
PKGLIST12 = [
   "logrotate",
   "libsqlite3-0",
   "venv-salt-minion",
]
```

File: susemanager/src/mgr_bootstrap_data.py [hunk 3]
```diff
@@ -109,36 +23,7 @@
 
 RES7_X86 = ["dmidecode"]
```
Comment (deneb-alpha) [inline susemanager/src/mgr_bootstrap_data.py:24]: same as before:

```suggestion
RES7_X86 = [
   "dmidecode",
]
```

File: susemanager/src/mgr_bootstrap_data.py [hunk 4]
```diff
@@ -109,36 +23,7 @@
...
-    "python3-pytz",
-    "python3-setuptools",
-    "python3-distro",
-    "python3-immutables",
-    "python3-contextvars",
-    "python3-looseversion",
-    "python3-jmespath",
-    "venv-salt-minion",
-]
+RES8 = ["venv-salt-minion"]
```
Comment (deneb-alpha) [inline susemanager/src/mgr_bootstrap_data.py:26]:
```suggestion
RES8 = [
   "venv-salt-minion",
]
```

File: susemanager/src/mgr_bootstrap_data.py [hunk 5]
```diff
@@ -109,36 +23,7 @@
...
 
 RES8_X86 = ["dmidecode"]
```
Comment (deneb-alpha) [inline susemanager/src/mgr_bootstrap_data.py:28]:
```suggestion
RES8_X86 = [
   "dmidecode",
]
```

File: susemanager/src/mgr_bootstrap_data.py [hunk 6]
```diff
@@ -247,117 +66,12 @@
...
-    "python-apt-common",
-    "python3-distro",
-    "python3-gnupg",
-    "gnupg",
-    "python3-immutables",
-    "python3-contextvars",
-    "python3-looseversion",
-    "python3-jmespath",
     "venv-salt-minion",
 ]
```
Comment (deneb-alpha) [inline susemanager/src/mgr_bootstrap_data.py:72]: Ubuntu 18.04 is EOL. I think we can drop it  completely. sorry. I can't add a  suggestion with a full drop

File: susemanager/src/mgr_bootstrap_data.py [hunk 7]
```diff
@@ -370,61 +84,9 @@
...
 
 PKGLISTDEBIAN9 = [
```
Comment (deneb-alpha) [inline susemanager/src/mgr_bootstrap_data.py:86]: Debian 9 is EOL too

File: susemanager/src/mgr_bootstrap_data.py [hunk 8]
```diff
@@ -433,10 +95,8 @@
 
 
 PKGLISTDEBIAN10 = [
```
Comment (deneb-alpha) [inline susemanager/src/mgr_bootstrap_data.py:97]: Debian 10 is  EOL

File: susemanager/src/mgr_bootstrap_data.py [hunk 9]
```diff
@@ -447,51 +107,17 @@
...
-    "salt-minion",
-    "gnupg",
+    # end of gnupg dependencies
     "venv-salt-minion",
-    "python3-gnupg",
-    "python3-looseversion",
-    "python3-jmespath",
 ]
 
 PKGLISTDEBIAN11 = [
```
Comment (deneb-alpha) [inline susemanager/src/mgr_bootstrap_data.py:123]: not sure what to do here. Debian 11 is in our grace period before EOL.

File: susemanager/src/mgr_bootstrap_data.py [hunk 10]
```diff
@@ -1050,260 +488,248 @@
...
+        "PKGLIST": PKGLIST15_SALT + PKGLIST15_X86_ARM,
         "DEST": DOCUMENT_ROOT + "/pub/repositories/sle/15/1/bootstrap/",
     },
     "SLES4SAP-15-SP1-ppc64le": {
         "PDID": [1770, 1710, 1765],
         "BETAPDID": [1926],
-        "PKGLIST": ONLYSLE15 + PKGLIST15_SALT + PKGLIST15_PPC,
+        "PKGLIST": PKGLIST15_SALT + PKGLIST15_PPC,
         "DEST": DOCUMENT_ROOT + "/pub/repositories/sle/15/1/bootstrap/",
     },
```
Comment (deneb-alpha) [inline susemanager/src/mgr_bootstrap_data.py:565]: do we still need 15 SP0 and SP1? to my knowledge, they should  be  gone

File: susemanager/src/mgr_bootstrap_data.py [hunk 11]
```diff
@@ -1396,61 +822,6 @@
...
-        "BASECHANNEL": "opensuse_leap15_3-x86_64",
-        "PKGLIST": PKGLIST15_SALT + PKGLIST15_X86_ARM,
-        "DEST": DOCUMENT_ROOT + "/pub/repositories/opensuse/15/3/bootstrap/",
-    },
-    "openSUSE-Leap-15.3-aarch64-uyuni": {
-        "BASECHANNEL": "opensuse_leap15_3-aarch64",
-        "PKGLIST": PKGLIST15_SALT + PKGLIST15_X86_ARM,
-        "DEST": DOCUMENT_ROOT + "/pub/repositories/opensuse/15/3/bootstrap/",
-    },
     "openSUSE-Leap-15.4-x86_64": {
```
Comment (deneb-alpha) [inline susemanager/src/mgr_bootstrap_data.py:825]: Leap 15.4 is gone too for one year now

File: susemanager/src/mgr_bootstrap_data.py [hunk 12]
```diff
@@ -1947,13 +1284,6 @@
...
         "DEST": DOCUMENT_ROOT + "/pub/repositories/rockylinux/9/bootstrap/",
     },
-    "ubuntu-16.04-amd64": {
-        "PDID": [-2, 1917],
-        "BETAPDID": [2061],
-        "PKGLIST": PKGLISTUBUNTU1604,
-        "DEST": DOCUMENT_ROOT + "/pub/repositories/ubuntu/16/4/bootstrap/",
-        "TYPE": "deb",
-    },
     "ubuntu-18.04-amd64": {
```
Comment (deneb-alpha) [inline susemanager/src/mgr_bootstrap_data.py:1287]: ubuntu 18.04 is gone

File: susemanager/src/mgr_bootstrap_data.py [hunk 13]
```diff
@@ -1982,12 +1312,6 @@
         "DEST": DOCUMENT_ROOT + "/pub/repositories/ubuntu/24/4/bootstrap/",
         "TYPE": "deb",
     },
-    "ubuntu-16.04-amd64-uyuni": {
-        "BASECHANNEL": "ubuntu-16.04-pool-amd64-uyuni",
-        "PKGLIST": PKGLISTUBUNTU1604,
-        "DEST": DOCUMENT_ROOT + "/pub/repositories/ubuntu/16/4/bootstrap/",
-        "TYPE": "deb",
-    },
     "ubuntu-18.04-amd64-uyuni": {
```
Comment (deneb-alpha) [inline susemanager/src/mgr_bootstrap_data.py:1315]: EOL

**Review by deneb-alpha — COMMENTED** [review]
Leaving a  summary here because  on some parts of the  file I wasn't able to properly drop a suggestion. 

* Debian 9 to 10 are EOL already, Debian  11 is currently in our grace period too
* Ubuntu 18.04 is EOL too. We only submit from 20.04 on
* For SLE, 15 SP0 and SP1 should be already EOL
* On the SLE 12 side, I admit I don't know if there are  special  contracts going  on, but to my  knowledge, everything older than 12-SP5 excluded could be finally dropped
* for SLES4SAP 11 SP4, I think there's still some support going on
* not sure if we want to drop older SUMA versions too. I think in the list I saw some 4.0 entries too
* Leap 15.4 is EOL too and 15.5 will be  EOL end of 2024.
* on some package lists I suggested to harmonize how we  write the list of packages. From what I  saw on past reviews, the way I suggested usually creates a better diff when we drop or add more packages, but of course, feel free to ignore the suggestion if you think it's not a good idea.

File: susemanager/src/mgr_bootstrap_data.py (hunk 2 above)
Comment (mcalmer) [reply susemanager/src/mgr_bootstrap_data.py:17]: The linter revert this.

File: susemanager/src/mgr_bootstrap_data.py (hunk 3 above)
Comment (mcalmer) [reply susemanager/src/mgr_bootstrap_data.py:24]: the linter revert this

File: susemanager/src/mgr_bootstrap_data.py (hunk 4 above)
Comment (mcalmer) [reply susemanager/src/mgr_bootstrap_data.py:26]: the linter revert this

File: susemanager/src/mgr_bootstrap_data.py (hunk 5 above)
Comment (mcalmer) [reply susemanager/src/mgr_bootstrap_data.py:28]: the linter revert this

File: susemanager/src/mgr_bootstrap_data.py (hunk 9 above)
Comment (mcalmer) [reply susemanager/src/mgr_bootstrap_data.py:123]: Then let's keep it for now

File: susemanager/src/mgr_bootstrap_data.py (hunk 10 above)
Comment (mcalmer) [reply susemanager/src/mgr_bootstrap_data.py:565]: Miguel said that as long as "SLES 15" is supported we should keep all SPs.
That's also the reason why we still have all SLES12 SPs in our list

File: susemanager/src/mgr_bootstrap_data.py (hunk 11 above)
Comment (mcalmer) [reply susemanager/src/mgr_bootstrap_data.py:825]: I am not sure if it was still used in one of our testsuites.

**Comment by mcalmer** [comment]
I will merge this now

**Pull request merged**